src/utils/requests.js

```diff
@@ -0,0 +1,51 @@
+import { useCallback } from "react";
+
+const base_url="https://localhost:8000";
+
+const sendRequest = useCallback(
+  async (url, method = "GET", body = null, headers = {}) => {
+    try {
+      const response = await fetch(base_url+ url, {
+        method,
+        body,
+        headers,
+      });
+      const responseData = await response.json();
+      if (!response.ok) throw new Error(responseData.status);
+    } catch (err) {
+      console.log(err);
+    }
+    return { responseData };
+  }
+);
+
+const login = (username,password) =>{
+    const res= await sendRequest("/users/auth/login/","POST",{username,password});
+    return res;
+}
+
+const getPostsbyStream = () =>{
+    const res=await sendRequest("/streams/<int:pk>/posts/");
```

Comment:
`<int:pk>` is a Django-related thing - it basically says you need to insert the id of the stream there. So here, take the stream id as an argument, and use a template string to insert it in place of `<int:pk>`. Same for fetchPostsByUser.